app/src/main/java/nextstep/signup/ui/component/PasswordConfirmTextField.kt

```diff
@@ -0,0 +1,70 @@
+package nextstep.signup.ui.component
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import nextstep.signup.R
+import nextstep.signup.domain.PasswordConfirmValidType
+import nextstep.signup.domain.PasswordConfirmValidator
+
+@Composable
+fun PasswordConfirmTextField(
+    modifier: Modifier = Modifier,
+    value: String,
+    onValueChange: (String) -> Unit,
+    passwordValue: String,
+) {
+    val supportingTextValue = if (value.isEmpty()) null
+    else when (PasswordConfirmValidator.match(passwordValue, value)) {
+        PasswordConfirmValidType.VALID -> null
+        PasswordConfirmValidType.INVALID_EQUAL -> stringResource(id = R.string.password_confirm_equal_error)
+    }
```

Comment:
if else와 when이 혼재되어있네요! 
아래 처럼 하나의 when으로 통일하면 어떨까요?

```suggestion
    val validationResult = PasswordConfirmValidator.match(passwordValue, value)
    val supportingTextValue = when {
        value.isEmpty() -> null
        validationResult is PasswordConfirmValidType.VALID -> null
        validationResult is PasswordConfirmValidType.INVALID_EQUAL -> stringResource(id = R.string.password_confirm_equal_error)
    }
```

Reply:
위 부분 관련하여 dm으로 질문 드렸습니다-!